bridge/bridge.go

```diff
@@ -362,9 +362,27 @@ func (b *Bridge) remove(containerId string, deregister bool) {
 				log.Println("removed:", containerId[:12], service.ID)
 			}
 		}
+		cleanUpAttrs := func(services []*Service) {
+		    b.registry.AcquireDistributedLock()
+            distServices, err := b.registry.DistributedServices()
+            if err != nil {
+                log.Println("Failed to list distributed services: ", err)
+            }
+			for _, service := range services {
+				if _, ok := distServices[service.Name]; ok {
+                    log.Println("Service exists in another container... Attrs will be kept:", service.Name)
+                } else {
+                    b.registry.RemoveAttributes(service)
+                    log.Println("Service was not found elsewhere... Attrs will be removed:", service.Name)
+                }
+			}
+			b.registry.ReleaseDistributedLock()
```

Comment:
**Why we need distributed lock?** 
- In some use cases of Consul, we have Consul running in quorum across multiple nodes. This is so we can keep track of multiple instances of services across the nodes and group them as one service instead of several.

When we have multiple instances of a service we choose to only remove the attributes of a service when the last instance is removed... thus if all instances are removed at one we can face a race condition in which the attributes are not removed. The distributed lock solves this problem